docs/designs/source-path-and-complement.md

```diff
@@ -0,0 +1,63 @@
+# Source Path
+
+In `.publish.json file`,`source_path` represents the source file's relative path to docset root.
+
+It will be used in PR comments to indicate the path of the files containing errors/warnings/suggestions.
+
+
+# Design of SourceMap
+
+For most cases, the same source_path corresponds to the same source file (i.e. One-To-One). And the source_path value is the source file's physical file path.
+
+For some special cases, it is allowed to config mappings to resolve source_path values, named **SourceMap**  
+
+|  key             |  Value            |
+|------------------|-------------------|
+| source file path | source_path value |
+
+Due to **SourceMap**, multiple source files may have the same source_path (i.e. **Many-To-One**). And the source_path value may be NOT equal to the source file's physical file path.
```

Comment:
What about service page which might come from either .md in repo, or generated .yml in memory. How are we going to address this scenario?

Reply:
Service pages are generated when build from multiple source files (i.e. Toc files). Virtual file paths are generated for them, refer to code https://github.com/dotnet/docfx/blob/0db0fef6a31cdb5a6cb366abc81922e9df201508/src/docfx/build/toc/ServicePageGenerator.cs#L82 and https://github.com/dotnet/docfx/blob/0db0fef6a31cdb5a6cb366abc81922e9df201508/src/docfx/build/toc/ServicePageGenerator.cs#L167

Reply:
So what will be the source_path and source_path_complement in such case?

Reply:
In such case, source_path  is the virtual path (which is unique), source_path_complement  is null.